client/cli/src/arg_enums.rs

```diff
@@ -163,6 +165,40 @@ impl Into<sc_service::config::RpcMethods> for RpcMethods {
 	}
 }
 
+impl RpcMethods {
+	/// Choose an RPC interface based on the RPC method chosen by the user and a few flags:
+	/// `is_external`, `is_unsafe_external` and `is_validator`.
+	pub fn choose_rpc_interface(
+		self,
+		is_external: bool,
+		is_unsafe_external: bool,
+		is_validator: bool,
+	) -> Result<IpAddr> {
+		if is_external && is_validator && self != RpcMethods::Unsafe {
+			return Err(Error::Input(
```

Comment:
We should return some custom error here, not a string. If someone uses this somewhere else, it is not set that these arguments are responsible for this error.

Reply:
Yes I had the same thought but then it's an "arg enum", an enum that is dedicated for CLI stuff with structopt/clap. I think it's fine. Probably the best would be to use anyhow instead of string errors